src/Repository/UserRepositoryInterface.php

```diff
@@ -0,0 +1,30 @@
+<?php
+/**
+ * Copyright (C) 2018 Adshares sp. z. o.o.
+ *
+ * This file is part of ADS Operator
+ *
+ * ADS Operator is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * ADS Operator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with ADS Operator.  If not, see <https://www.gnu.org/licenses/>
+ */
+
+namespace Adshares\AdsOperator\Repository;
+
+use Adshares\AdsOperator\Document\User;
+
+interface UserRepositoryInterface
+{
+    public function findByEmail(string $email):? User;
```

Comment:
I would remove `?` and return a null object implementing the same interface as user
OR
throw an exception - which is not perfect but better than null

Reply:
I know the idea (Null object pattern) but I'm not a fun of it. Throwing exceptions in such situations sounds ok for me. To be honest this method is not needed anymore because I started using doctrine validator for checking unique values.
I leave it for one reason only - it will be needed in the future to get a user.